controllers/attributeObserver/htmlAttributeObserverController.js

```diff
@@ -14,9 +14,9 @@ export class HtmlAttributeObserverController {
 		);
 
 		this._host = host;
-		host.addController(this);
 		this._attributes = attributes;
 		this.values = new Map();
+		host.addController(this);
```

Comment:
I'm going to re-test to see if this reordering is still necessary, or if it's something I can just drop because I'm being more careful about order-of-operations elsewhere.

Reply:
Ah, I know why this is necessary now. Turns out reactive controllers _can_ be hooked up after the host is connected, but if that happens Lit immediately calls `hostConnected` when the controller is added to the host. With the new async workflow, the controller now instantiates after the host is connected and the previous code explodes because `this._attributes` is undefined.